bdb/bdb_osqltrn.c

```diff
@@ -328,24 +337,24 @@ bdb_osql_trn_t *bdb_osql_trn_register(bdb_state_type *bdb_state,
         }
     }
 
-    Pthread_mutex_lock(&trn_repo_mtx);
-
-    if (!trn_repo)
-        goto done;
-
     trn = (bdb_osql_trn_t *)calloc(1, sizeof(bdb_osql_trn_t));
     if (!trn) {
         *bdberr = BDBERR_MALLOC;
-        goto done;
+        return NULL;
     }
 
     Pthread_mutex_init(&trn->log_mtx, NULL);
-
     trn->shadow_tran = shadow_tran;
-    trn->trak = (trak & SQL_DBG_BDBTRN) || (trn_repo->trak);
-
     listc_init(&trn->bkfill_list, offsetof(bdb_osql_log_t, lnk));
 
+    Pthread_mutex_lock(&trn_repo_mtx);
+    if (!trn_repo) {
+        cleanup_trn(trn);
+        return NULL;
+    }
```

Comment:
I'm a bit confused here.  When is this lock on trn_repo_mtx unlocked?

Reply:
Bah, fixed. Was too hasty. Good news is it does not get triggered unless trn_repo is null which is at exit only.
Thanks for catching.